### PR TITLE
MHER-53: Debian packaging scripts - initial commit

### PR DESCRIPTION
**Pull request by BMartinos**
Added scripts to allow for the mHero mediator to be packages in a debian file

### CODE REVIEW

**Comment by BMartinos** [comment]
Hey @rcrichton Ive gotten quite far with the deb packaging but running into a weird error which im not sure what the problem is. In the control file ive added two dependencies and even though they are installed, the package installation fails because it does not find the dependencies. Have you experienced something like this before?

File: targets/trusty/debian/control [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Section: web
+Priority: optional
+Standards-Version: 3.9.1
+Build-Depends: debhelper (>= 7)
+Homepage: https://github.com/jembi/openhim-mediator-mhero
+
+
+Package: openhim-mediator-mhero
+Architecture: amd64
+Depends: openhim-core, nvm
```
Comment (rcrichton) [inline targets/trusty/debian/control]: I think this actually need to be `openhim-core-js`. Also I'm not sure if there is a `nvm` package in the ubuntu repositories. Either way, this should be installed by the `openhim-core-js` package I think.

**Comment by BMartinos** [comment]
Hey @rcrichton I have added the necessary scripts and was able to successfully install in a clean vagrant instance. If you can review this for me when you get a chance. Thanks

**Comment by rcrichton** [comment]
@BMartinos I'm just starting to review this now, but one quick first comment. We usually add all packaging file and folders to a `packaging` folder. Would you be able to do  that here to? I just keep it separate out from the source code.

File: targets/trusty/debian/changelog [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+ -- Martin Brocker <martin@jembi.org>  Fri, 29 Apr 2016 15:35:23 +0200
+
+openhim-mediator-mhero (0.1.0-1~trusty) trusty; urgency=medium
+
+  * Release Debian Build 0.1.0-1.
+
+ -- Martin Brocker <martin@jembi.org>  Fri, 29 Apr 2016 15:33:24 +0200
+
+openhim-mediator-mhero (0.1.0-1~trusty) trusty; urgency=low
```
Comment (rcrichton) [inline targets/trusty/debian/changelog]: We need to reset this back to the first entry as most of these builds are just testing builds. We should only commit the changelog when we upload the package.

File: targets/trusty/usr/share/openhim-mediator-mhero [hunk 3]
```diff
@@ -0,0 +1 @@
+Subproject commit 1420fd4cb809e0f1fcdf6f20d0b8a3a6c79fcee6
```
Comment (rcrichton) [inline targets/trusty/usr/share/openhim-mediator-mhero]: What is this?

**Comment by rcrichton** [comment]
Perhaps also add a brief readme text in the packaging folder to tell someone how to go about the packaging.

File: cp-mediators-into-pkg.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -e
+
+SRCDIR="/home/martin/git"
```
Comment (rcrichton) [inline cp-mediators-into-pkg.sh]: This isn't used. (May have been un-used in my script to)

**Comment by rcrichton** [comment]
@BMartinos when I install this it seems to work well great however I'm having a problem with the mediator config.

It gets set to `"apiURL": "localhost:8080"` where it should be `"apiURL": "https://localhost:8080"`. I could be doing something wrong.

**Comment by BMartinos** [comment]
Hi @rcrichton I have made the corrections you mentioned. I not sure what that subproject commit was on the main repository. I think this might have been auto generated when I created my branch. If you can review it again for me please. Thanks

**Comment by rcrichton** [comment]
Thanks @BMartinos look good. I've added a few things to this including the Vagrantfile I used for testing.

**Pull request merged**